### DISCUSSION TITLE
`forwarded_allow_ips` should allow wildcard IP addresses

### DISCUSSION THREAD

**Discussion by Shiakaron** [Kludex/uvicorn#2358] [Ideas]
In the use case of kubernetes, where the reverse proxy pods can have any IP address looking like `192.168.*.*`. Therefore it would make sense to allow adding IP addresses with wildcards in the `forwarded_allow_ips` list, instead of specifying `*` to trust everything, or passing in a list of 254^2 (~65000) addresses. 

All that would be required is a little tweaking in the `ProxyHeadersMiddleware.get_trusted_client_host` method

**Comment by toofishes** [comment]
After some recent changes (#2468), you can use network specifiers as well, after looking at the code: https://github.com/encode/uvicorn/blob/master/uvicorn/middleware/proxy_headers.py#L67-L75

So something like `FORWARDED_ALLOW_IPS=192.168.0.0/16` would work for your example.

**Reply by BloodyRain2k** [reply]
That does not seem to work for me using Uvicorn 0.34.2.
I am trying to run a FastAPI server in a docker and get it to trust my reverse proxy Nginx in the same network segment.
Both are in `172.*` (172.24. for Uvicorn, 172.22. for the RP), but it doesn't matter if I start uvicorn (in the docker-entry.sh) with `--forwarded-allow-ips '172.0.0.0/24'` or `--forwarded-allow-ips '0.0.0.0/32'`, it's still breaking the 307s returned by FastAPI. Only when I set it to `--forwarded-allow-ips '*'` do they work.
